prediction_market_agent_tooling/tools/perplexity/perplexity_client.py

```diff
@@ -0,0 +1,86 @@
+from typing import Any, Dict, List, Optional
+
+import httpx
+from pydantic import SecretStr
+
+from prediction_market_agent_tooling.tools.perplexity.perplexity_models import (
+    PerplexityModelSettings,
+    PerplexityRequestParameters,
+    PerplexityResponse,
+)
+
+
+class PerplexityModel:
+    def __init__(
+        self,
+        model_name: str,
+        *,
+        api_key: SecretStr,
+        completition_endpoint: str = "https://api.perplexity.ai/chat/completions",
+    ) -> None:
+        self.model_name: str = model_name
+        self.api_key: SecretStr = api_key
+        self.completition_endpoint: str = completition_endpoint
```

Comment:
This isn't a problem, but you don't have to overtype it, unless you prefer it. Just

```suggestion
        self.completition_endpoint = completition_endpoint
```

Would work fine. (similarly for others)

Reply:
Done